system/jlib/jtrace.cpp

```diff
@@ -1591,7 +1591,11 @@ ActiveSpanScope::ActiveSpanScope(ISpan * _ptr, ISpan * _prev) : span(_ptr), prev
 ActiveSpanScope::~ActiveSpanScope()
 {
     ISpan* current = queryThreadedActiveSpan();
-    if (current != span)
+
+    // NOTE: need to handle span == nullptr differently because queryThreadActiveSpan returns the cNullSpan in this case 
```

Comment:
When was this occurring?  Can you provide some more details - because I would expect span to never be null.

Reply:
@ghalliday The only place I am aware of is here in the dafilesrv code: https://github.com/hpcc-systems/HPCC-Platform/blob/master/fs/dafsserver/dafsserver.cpp#L1127

Reply:
@ghalliday thoughts on the above? I have been looking at ActiveSpanScope(nullptr) as a valid use case to make it easier for developers to handle code that may or may not be traced. However, I can see an argument for defaulting to CNullSpan in these cases.

Reply:
No, I think that is a better approach. I will make that change